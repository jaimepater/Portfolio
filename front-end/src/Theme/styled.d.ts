import 'styled-components';

// and extend them!
declare module 'styled-components' {
  export interface DefaultTheme {
    palette: {
      blue1: string,
      white: string,
      gray1: string,
    }
    fontSize: {
      xl: string,
      lg: string,
      md: string,
      sm: string,
      xs: string,
    }
  }
}
