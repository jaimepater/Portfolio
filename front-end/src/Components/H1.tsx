import styled from 'styled-components';

const H1 = styled.h1`
  font-size: ${props => props.theme.fontSize.xl};
  font-weight: normal;
`;

export default H1;
