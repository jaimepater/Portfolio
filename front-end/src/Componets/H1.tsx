import * as React from 'react';
import styled from 'styled-components';


const H1 = styled.h1`
 font-size: ${props => props.theme.fontSize.xl};
`;

export default H1;
