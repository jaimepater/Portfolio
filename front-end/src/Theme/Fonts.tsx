import { css } from 'styled-components';
import Italianno from '../assets/Fonts/rageitalic/rageitalic.woff2';
import mexcellent from '../assets/Fonts/Mexcellent3D-Regular/mexcellent3d.woff2';

export const FontFaces = css`
 @font-face {
  font-family: 'Italianno';
  font-style: normal;
  font-weight: 400;
  src: url(${Italianno}) format('woff2');
}
`;

export const Mexcellent = css`
 @font-face {
  font-family: 'mexcellent';
  font-style: normal;
  font-weight: 400;
  src: url(${mexcellent}) format('woff2');
}
`;

export const FontSize = {
  xl: '50px',
  lg: '30px',
  md: '16px',
  sm: '14px',
  xs: '12px',
};
