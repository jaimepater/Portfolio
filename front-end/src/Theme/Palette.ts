const Palette = {
  white: '#FFFFFF',
  blue1: '#1592E6',
  gray1: '#707070',

};


export default Palette;
