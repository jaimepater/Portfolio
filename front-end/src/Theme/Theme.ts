import { DefaultTheme } from 'styled-components';
import Palette from './Palette';

export const Theme: DefaultTheme = {
  palette: Palette,
};

export default Theme;
