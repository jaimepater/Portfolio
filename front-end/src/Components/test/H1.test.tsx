import React from "react";
import { shallow } from 'enzyme';
import H1 from "../H1";
import {Theme} from "../../Theme/Theme";

describe('H1', () => {
  it('Renders H1', () => {
    const shallowComponent = shallow(<H1 theme={Theme} />);
    expect(shallowComponent).toMatchSnapshot();
  });
});