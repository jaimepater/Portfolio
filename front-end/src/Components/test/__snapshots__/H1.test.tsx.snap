// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 Renders H1 1`] = `
<h1
  className="sc-AykKC ctpwKh"
/>
`;
